### PR TITLE
chore(Platform): Dockerize app and migrate to AWS 

### LINKED ISSUE #131: Migrate from Heroku to AWS

### PR DESCRIPTION
**Pull request by mapra99**
This PR adds some required stuff to dockerize the application and migrate from Heroku to AWS. The following changes were made:
- A Dockerfile is added, to build the app on a ruby alpine container.
- Docker-compose config files are added for development and testing environments
- A new environment called `staging` is created
- Some env var names changes and some dependencies config such as the database, Redis, and Sidekiq
- A very simple `/health` endpoint is added to associate it to the health check config on aws.
- README is updated with instructions to mount the project with docker

### CODE REVIEW

File: config/credentials.yml.enc [hunk 1]
```diff
@@ -1 +1 @@
-4rh+iJhXDyXAnCzz3JjexT+kLBVuQ3Q3ILMmyZUUiMoQOBCG3u1Vi7r7gQJn7caNjkBF5voem6o4PKei9t78TlHm3BFdAL0H6dlfYR4/6GWDDih2p0jGCSiUGcTTwZA1IutI7Y00/557w10oNqh5la2gEi4w/i5dfNXBaH212G7n9uNgucgPftRQsjGIux3LJ7/v3hoUogVdpntIggmTt45tmXkknHQWNASnrTfpw39iLTqeU6X9w640majeKKPDSMg4J4RMxrwGmFtoPVKvYXqHPQ98NvBTX5GLcq5ltPnLlZGM6BzewOh/f3CBT7v19nXLPA7GC1h+LiXASDKHnAyYEsDRJ7xAmVDPOw0T9huqwXTvWGavvi/F210y+44KDO14SMYp1e6+3mIhYOPb3MyDTAi0pjaMlCBw--aPb8kUihaMa60XgH--QZWMXfNr4obg4IoLtdVc5A==
\ No newline at end of file
+1Od3l3/ZUyBmZGnqQY8b8g5jSa016+iuHigiRRhYU3oaqjTQh5op131Q/ydxVsp4fu0nsG5Ocnr+VYeR1LGXONrY50rLVLV0l3AFzNota7XwHZ7wRFjjLO6EKcjc/4ddjnGnfMrfeZ0ciXTUYVjJJR52yJQw/h0ZhrlCX9o0S4cLm8b4+9YbsA+xVyw7jdFzf+FR/ogSOL/QFGwvVm8+KkpgCUrpIpPOX4/ULrZ2BSB4MqXPKFtIG+O6xjppRQ0TpPyf5+HCChOtAATxK8NtgBMy+jKLleOUR2CpvE38JPHSMUa/O1dKFs8q2Fv60PTlMZgIU7H7kdkTcEpxtyfscewAtSFhjf8LF7C6Aqoe4R+ltHWvU1E8tINFMNdihuuyrWa4s5cL5CPESty093sf0VLZCLeKKkDe79Up--SenOYOnVom0mJbWd--bg3ZeLUKh9ljqUuXuBDd/Q==
```
Comment (ImMPrada) [inline config/credentials.yml.enc:1]: EOF or because it is encrypted should not have EOF?

File: config/initializers/geocoder.rb [hunk 2]
```diff
@@ -34,12 +34,9 @@
...
   cache: Redis.new(
-    url: if Rails.env.production?
-           ENV.fetch('REDISCLOUD_URL')
-         else
-           ENV.fetch('REDIS_URL',
-                     'redis://localhost:6379')
-         end
+    url: ENV.fetch('REDIS_URL', 'redis://redis:6379'),
+    username: ENV.fetch('REDIS_USERNAME', nil),
+    password: ENV.fetch('REDIS_PASSWORD', nil)
```
Comment (ImMPrada) [inline config/initializers/geocoder.rb:39]: Like it!

**Review by ImMPrada — APPROVED** [review]
Just one comment and that's all

File: config/credentials.yml.enc (hunk 1 above)
Comment (mapra99) [reply config/credentials.yml.enc:1]: I'm not sure if adding the EOF would break it 😅. This file is generated by rails automatically

**Pull request merged**